src/data/DDTable.tsx

```diff
@@ -55,70 +57,92 @@ export function getSearchableColumns(columns: Column[]): SearchableColumn[] {
 
 export default function DDTable({ payload, title, props }: DDComponentProps) {
   const allRows = payload as { rowData: any[] }[]
-  const initialState: TableState = {
-    rows: allRows,
-    motivationFilter: [],
-    motivationThreshold: 4,
-    competenceFilter: [],
-    competenceThreshold: 3,
+  const initialState = {
+    filters: [
+      {
+        name: 'COMPETENCE',
+        values: [],
+        threshold: 3,
+        placeholder: 'Filtrer på kompetanse...',
+        datafetch: useCategories,
+      },
+      {
+        name: 'MOTIVATION',
+        values: [],
+        threshold: 4,
+        placeholder: 'Filtrer på motivasjon...',
+        datafetch: useCategories,
+      },
+      {
+        name: 'CUSTOMER',
+        values: [],
+        placeholder: 'Filtrer på kunder...',
+        threshold: 0,
+        datafetch: useCustomer,
+      },
+    ],
     searchTerm: '',
   }
-  const [state, dispatch] = useReducer(reducer, initialState)
+  const [state, setState] = useState<TableState>(initialState as TableState)
```

Comment:
Tror det kan bli ryddigere å ha en state hver for filters og searchTerm. De brukes for det meste separat. Det samme gjelder i CustomerFilter.

Reply:
Er enig med å flytte searchterm ut, men filtrene så er det lettere å ha de samlet i en array, da du slipper ekstra håndtering i senere tid ved implementasjon av flere filter.

Reply:
Er det jeg mener. Ser for meg noe sånt:
```suggestion
  const [filters, setFilters] = useState<FilterObject[]>(initialFilters)
  const [searchTerm, setSearchTerm] = useState('')
```